web_ui/packages/ui/src/button/button.component.tsx

```diff
@@ -44,5 +64,8 @@ export const Button = forwardRef((props: ButtonProps, ref: ButtonProps['ref']) =
 });
 
 export const ActionButton = forwardRef((props: ActionButtonProps, ref: ActionButtonProps['ref']) => {
-    return <SpectrumActionButton {...props} ref={ref} />;
+    const { colorMode = ColorMode.DARK, UNSAFE_className, ...rest } = props;
+    const className = getActionButtonClass(colorMode, UNSAFE_className);
+
+    return <SpectrumActionButton {...rest} ref={ref} UNSAFE_className={className} />;
```

Comment:
This approach ensures that `getActionButtonClass` has a single responsibility and also removes the need for the nested ternary condition

```suggestion
    return <SpectrumActionButton {...rest} ref={ref} UNSAFE_className={clsx(UNSAFE_className, getActionButtonClass(colorMode))} />;
```

Reply:
I like it. Updated!